docs/guide/dev/integration/react/v2/README.md

```diff
@@ -0,0 +1,526 @@
+
+```js
+const { editor, status, error, loading } = useCKEditor( {
+	config: configObject,
+	debug: true,
+	dispatchEvent: ( { type, payload } ) => {
+		if ( type === CKEditorEventAction.instanceReady ) {
+			alert( 'Editor is ready!' );
+		}
+	},
+	editorUrl: 'https://your-website.example/ckeditor/ckeditor.js',
+	element: elementObject,
+	type: 'classic',
+	subscribeTo: [ 'instanceReady' ]
+} );
+```
+
+### dispatchEvent explained
+
+By providing `dispatchEvent` function, the `useCKEditor` hook's API unlocks more powerful patterns than `CKEditor` component has to offer.
+
+For instance, pass `dispatch` from `useReducer` in order to listen to editor events and derive state of your components as needed. Please note that reducer function of `useReducer` hook is not a suitable place to perform side-effects.
+
+In the example below, `reducer` is used to calculate next state of the component.
+
+```js
+import React from 'react';
+import { CKEditorEventAction, useCKEditor } from 'ckeditor4-react';
+
+function Editor( { dispatchEvent, initData } ) {
+	// Use `useState` rather than `useRef` in order to trigger re-render.
+	const [ element, setElement ] = React.useState();
+
+	const { status } = useCKEditor( {
+		element,
+		dispatchEvent,
+		subscribeTo: [ 'blur', 'change', 'focus' ]
+	} );
+
+	return (
+		<div
+			ref={setElement}
+			style={status !== 'ready' ? { visibility: 'hidden' } : undefined}
+		>
+			{initData}
+		</div>
+	);
+}
+
+function Feedback() {
+	// `dispatch` can be passed directly to `useCKEditor` hook
+	const [
+		{ canSendFeedback, data, isUserEditing },
+		dispatch
+	] = React.useReducer( reducer, {
+		canSendFeedback: false,
+		data: undefined,
+		isUserEditing: false
+	} );
+
+	const handleClick = () => {
+		alert( `Feedback has been submitted successfully:\n${ data }` );
+	};
+
+	return (
+		<div>
+			<Editor
+				dispatchEvent={dispatch}
+				initData={<p>Let us hear your feedback!</p>}
+			/>
+			<button disabled={!canSendFeedback} onClick={handleClick}>
+				Send
+			</button>
+			{isUserEditing && <div>We're happy to hear your feedback!</div>}
+		</div>
+	);
+}
+
+function reducer( state, action ) {
+	switch ( action.type ) {
+		case CKEditorEventAction.change:
+			const data = action.payload.editor.getData().trim();
+			return {
+				...state,
+				canSendFeedback: !!data,
+				data
+			};
+		case CKEditorEventAction.focus:
+			return {
+				...state,
+				isUserEditing: true
+			};
+		case CKEditorEventAction.blur:
+			return {
+				...state,
+				isUserEditing: false
+			};
+		default:
+			return state;
+	}
+}
+
+export default Feedback;
+```
+
+For more examples see [repo](https://github.com/ckeditor/ckeditor4-react/tree/master/samples).
```

Comment:
After the release of v2, the newest samples will be on `stable` branch. We should redirect users to the stable version of samples.
```suggestion
For more examples see [repo](https://github.com/ckeditor/ckeditor4-react/tree/stable/samples).
```